### PR TITLE
fix(host_metrics source):  `logical_cpus` and `physical_cpus` separate gauges

### LINKED ISSUE #14322: Split up `cpus` gauge into `logical_cpus` and `physical_cpus`
In https://github.com/vectordotdev/vector/pull/13874 we added a new gauge for number of CPUs but modeled it as a single gauge with a `mode` tag to differentiate between physical and logical CPUs. Given that users will generally never want to use these two gauges together (e.g. summing them) I think it'd be better to split them out into two gauges:

* `logical_cpus`
* `physical_cpus`

We should add these new gauges and deprecate the `cpus` gauge.

### PR DESCRIPTION
**Pull request by neuronull**
Fixes: #14322.

### CODE REVIEW

**Comment by spencergilbert** [comment]
This looks like it drops the current metric, I think we need to keep it for a deprecation period.

**Comment by neuronull** [comment]
> This looks like it drops the current metric, I think we need to keep it for a deprecation period.

There was a convo with @jszwedko who suggested that we could skip deprecation given that it is a bug.

**Comment by spencergilbert** [comment]
> There was a convo with @jszwedko who suggested that we could skip deprecation given that it is a bug.

Missed that! Just saw the issue opened.

File: src/sources/host_metrics/cpu.rs [hunk 1]
```diff
@@ -105,15 +97,40 @@ mod tests {
...
+        }
 
-        // cpu count metrics should be present
-        let mut iter = metrics.iter();
-        assert!(iter.any(|metric| { metric.tag_matches(MODE, LOGICAL_CPUS) }));
-        assert!(iter.any(|metric| { metric.tag_matches(MODE, PHYSICAL_CPUS) }));
+        // cpu count metrics should each be present once
+        assert_eq!(
+            n_logical_cpus, 1,
+            "There can only be one! (logical_cpus metric)"
```
Comment (spencergilbert) [inline src/sources/host_metrics/cpu.rs:129]: HIGHLANDER

**Review by spencergilbert — APPROVED** [review]

**Review by bruceg — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Tested this locally:

```
{"name":"logical_cpus","namespace":"host","tags":{"collector":"cpu","host":"a-8cktst39ij0a"},"timestamp":"2022-09-08T15:00:51.921434072Z","kind":"absolute","gauge":{"value":8.0}}
{"name":"physical_cpus","namespace":"host","tags":{"collector":"cpu","host":"a-8cktst39ij0a"},"timestamp":"2022-09-08T15:00:51.921434072Z","kind":"absolute","gauge":{"value":4.0}}
```

**Comment by jszwedko** [comment]
Just realizing we neglected to add these new metrics to the cue docs.

**Comment by neuronull** [comment]
> Just realizing we neglected to add these new metrics to the cue docs.

Opened https://github.com/vectordotdev/vector/pull/14366

**Pull request merged**